### PR TITLE
draft - samehex sign in

### PR DESCRIPTION
**Pull request by mhagel**
## Link to Issue
Closes: #5078 
Closes: #5226 

## Description of Changes
- Adds FIXME widget to TODO page.
- 

## Test Plan
- Unit tested the `FIXME()` call.
- CA (click around) tested on local and frack:
  - TODO page
  - 

## Deployment Plan

1. 

## Other Considerations

-

### CODE REVIEW

**Comment by mhagel** [comment]
Progress note: I thought this might be able to get merged without a big migration of existing data, but now I'm not sure. Right now I am experimenting with the big migration, and trying to figure out how to undo all the data changes in the `down` function for safety.

File: packages/commonwealth/server/migrations/20231002202025-cosmos-hex.js [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+          WHERE hex = '${hex.hex}';
+          `,
+          { transaction: t }
+        );
+      }
+    });
+  },
+
+  down: async (queryInterface, Sequelize) => {
+    await queryInterface.sequelize.transaction(async (t) => {
```
Comment (mhagel) [inline packages/commonwealth/server/migrations/20231002202025-cosmos-hex.js:84]: @timolegros As you can see, above we alter a lot of data. Changing `user_id`s and `profile_id`s. Can you imagine a way to undo these changes in the `down` function? Or do we rely on a backup DB in this type of case?

Comment (timolegros) [reply packages/commonwealth/server/migrations/20231002202025-cosmos-hex.js:84]: If the `up` function is removing data/record relationships then those can't be recovered in the `down` function. Ideally, for changes like this, we use the [expand-contract pattern][1] which essentially just means that the old data still exists (we don't overwrite it) and simply drop it when the changes are confirmed.

That said, we have rarely ever done this since as far as I can remember we haven't needed to use the `down` function so it might be sufficient to say that we can use a database dump in the absolute worst-case scenario. While we really should get into a habit of writing reversible migrations, this takes a lot of time and isn't a priority at this time. My general rule of thumb up to this point has been I make a migration reversible if it takes me a reasonable amount of time (1-2 points but no more than that).

IMO it should be fine to leave `down` empty. I literally just did this [here][2].

I would like to note though that your approach to the migration might cause issues in production. You are performing thousands upon thousands of individual `UPDATE` queries within a single transaction (using for loop) which can lead to locking/performance issues. Let's hop on a call tomorrow and discuss query alternatives + explore if we can easily apply the expand-contract pattern.

[1]: https://medium.com/continuousdelivery/expand-contract-pattern-and-continuous-delivery-of-databases-4cfa00c23d2e
[2]: https://github.com/hicommonwealth/commonwealth/pull/5207/files#diff-d5cfbbb1b156adc70989778fe827cc2d04c497d4925685f6071f2d6bea4b208aR35

Comment (mhagel) [reply packages/commonwealth/server/migrations/20231002202025-cosmos-hex.js:84]: Thanks. I'll read up on expand-contract.

It is a massive migration, and I'm open to suggestions. Maybe we can break it into separate transactions.

File: packages/commonwealth/server/migrations/20231005213247-assign_master_profile.js [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+          FROM MasterProfiles MP
+          WHERE A.hex = MP.hex;
+          `,
+          { transaction: t }
+        );
+      } catch (e) {
+        // console.log('error', e);
+        throw new Error(e);
+      }
+    });
```
Comment (mhagel) [inline packages/commonwealth/server/migrations/20231005213247-assign_master_profile.js:61]: @timolegros This is a working example of using the Profile with most-populated fields.

I have been trying your idea of making a new profile that combines populated fields from existing profiles, favoring the latest update. But so far I have failed due to the complexity. Do you feel like taking a crack at it or finding time to pair on it? 

We can also talk about it in OH.

Comment (timolegros) [reply packages/commonwealth/server/migrations/20231005213247-assign_master_profile.js:61]: Let's discuss in OH and pair after if needed.

Comment (timolegros) [reply packages/commonwealth/server/migrations/20231005213247-assign_master_profile.js:61]: To note though you don't have to do absolutely everything in a single step. You shouldn't make queries in a JS loop but that doesn't mean you can't pull all the profile data into JS, consolidate the data (create the master profile objects), then submit a single bulkCreate operation to create all the master profiles, and then finally link the master profiles to addresses.

Comment (mhagel) [reply packages/commonwealth/server/migrations/20231005213247-assign_master_profile.js:61]: Can you do a bulkCreate in a migration? I don't see any precedent.

Comment (mhagel) [reply packages/commonwealth/server/migrations/20231005213247-assign_master_profile.js:61]: nvm I see bulkInsert

**Comment by mhagel** [comment]
Closing for now. Specific AC has changed from when this was drafted.
